server/app/views/errors/ErrorComponent.java

```diff
@@ -0,0 +1,77 @@
+package views.errors;
+
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.li;
+import static j2html.TagCreator.p;
+import static j2html.TagCreator.rawHtml;
+import static j2html.TagCreator.ul;
+
+import j2html.tags.UnescapedText;
+import j2html.tags.specialized.DivTag;
+import java.util.Optional;
+
+/**
+ * Render an error component based on the USWDS 404 page template
+ *
+ * @see <a
+ *     href="https://designsystem.digital.gov/templates/404-page">https://designsystem.digital.gov/templates/404-page</a>
+ */
+public final class ErrorComponent {
+
+  public static DivTag renderErrorComponent(
+      String title,
+      Optional<String> subtitle,
+      Optional<UnescapedText> additionalInfo,
+      String buttonText,
+      String buttonLink,
+      Optional<String> statusCode) {
+
+    DivTag button =
+        div()
+            .withClass("margin-y-5")
+            .with(
+                ul().withClass("usa-button-group")
+                    .with(
+                        li().withClass("usa-button-group__item")
+                            .with(
+                                a().withClass("usa-button")
+                                    .withId("visit-home-page-button")
+                                    .withText(buttonText)
+                                    .withHref(buttonLink))));
```

Comment:
**question:** Would this ever be something different than `messages.at(MessageKey.BUTTON_HOME_PAGE.getKeyName());` and `routes.HomeController.index().url();`? If so, we can reduce the arguments that need to be passed and have those defined in this class?

Reply:
The button text will change (it will be [this](https://github.com/civiform/civiform/blob/dd6e5e2afc7350a8fe83e2cb999a64c97cb44190/server/conf/i18n/messages#L797) for this button once translations come back). I wanted to leave the link changeable as well, but now that I'm thinking about it, I'm not sure what else it would be. I'll remove the link param.